gammapy/scripts/image_pipe.py

```diff
@@ -175,13 +203,132 @@ def make_maps(self, radius, bkg_norm=True):
+
+    def make_exposure_weighted_spectrum(self, obs_id, spectral_index=2.3):
+        """Compute the exposure map for one observation. Excess/exposure will give the integrated flux
+        over the self.energy_band
+
+        Parameters
+        ----------
+        obs_id : int
+            Number of the observation
+        spectral_index : int
+
+        """
+        ##TODO: should be re-implemented using the exposure_cube function and the
+        # spectral_cube.integral_flux_image method
+        exposure = SkyMap.empty_like(self.empty_image)
+
+        # Determine offset value for each pixel of the map
+        xpix_coord_grid, ypix_coord_grid = exposure.coordinates(coord_type='pix')
+        # calculate pixel offset from center (in world coordinates)
+        coord = pixel_to_skycoord(xpix_coord_grid, ypix_coord_grid, exposure.wcs, origin=0)
+        center = SkyCoord.from_pixel(self.header["NAXIS1"] / 2., self.header["NAXIS2"] / 2., exposure.wcs)
+        offset = coord.separation(center)
+
+        obs = self.data_store.obs(obs_id=obs_id)
+        livetime = obs.observation_live_time_duration
+
+        # 2D Exposure computation on the self.energy_range and on an offset_tab
+        energy = EnergyBounds(np.linspace(self.energy_band[0].value, self.energy_band[1].value, 100),
+                              self.energy_band.unit)
+        energy_band = energy.bands
+        energy_bin = energy.lin_centers
+        eref = EnergyBounds(self.energy_band).log_centers
+        spectrum = (energy_bin / eref) ** (-spectral_index)
+        aeff2d = obs.aeff
+        offset_tab = Angle(np.linspace(self.offset_band[0].value, self.offset_band[1].value, 10), self.offset_band.unit)
+        exposure_tab = np.sum(aeff2d.evaluate(offset_tab, energy_bin).to("cm2") * spectrum * energy_band, axis=1)
+        # Divide by the integrated spectrum
+        norm = np.sum(spectrum * energy_band)
+        exposure_tab /= norm
+
+        # Interpolate for the offset of each pixel
+        f = interpolate.interp1d(offset_tab, exposure_tab, bounds_error=False, fill_value=0)
+        exposure.data = f(offset)
+        exposure.data *= livetime
+        exposure.data[offset >= self.offset_band[1]] = 0
+        self.maps["exposure"] = exposure
+
+    def make_total_exposure(self, spectral_index=2.3, exposure_weighted_spectrum=False):
+        """Compute the exposure map for all the observations in the obs_table.
+
+        Parameters
+        ----------
+        spectral_index : int
+            spectral index of the source spectrum
+        exposure_weighted_spectrum : bool
+            True if you want that the total excess / exposure gives the integrated flux
+        """
+        exposure_map = SkyMap.empty_like(self.empty_image)
+        for obs_id in self.obs_table['OBS_ID']:
+            if self._has_counts(obs_id) == 0:
+                self.maps["exposure"].data[:] = 0
+            else:
+                if exposure_weighted_spectrum:
```

Comment:
I'd have to think about what a good API is (like this or separate methods).
No time. If this works for you, let's just leave as-is for now?